### ISSUE TITLE
Issue with UnusedImports and javadoc usage.

### ISSUE THREAD

**Issue by liutikas** [checkstyle/checkstyle#3453]
$ javac Test.java 
`[no output]
`

$ cat Test.java

```
package foo.bar;

import java.lang.Thread;

/**
 * Use {@link Thread.State} in this javadoc.
 */
public class Test {
}
```

$ cat config.xml

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="UnusedImports">
            <property name="processJavadoc" value="true"/>
        </module>
    </module>
</module>
```

$ java -jar checkstyle-7.2-SNAPSHOT-all.jar -c config.xml Test.java

```
Starting audit...
[ERROR] /usr/local/google/ssd1/checkstyle/target/Test.java:3:8: Unused import - java.lang.Thread. [UnusedImports]
Audit done.
Checkstyle ends with 1 errors.
```

---

I expected no errors as the import is used by the Javadoc.

---

This happens on the latest master of Checkstyle as well as older versions like 6.12.1
##

**Comment by MEZk** [comment]
@romani 
It is a FP since javadoc links can require import.

**Comment by romani** [comment]
@MEZk , please make an issue on DetailAST cache problem

**Comment by romani** [comment]
This issue is due to fact that this Check is using regexp parsing for javadoc, but it should use javadoc parser http://checkstyle.sourceforge.net/writingjavadocchecks.html

**Comment by romani** [comment]
This Check is unfortunate implementation based on regexp parsing of javadoc
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java
attention to `.... extends AbstractCheck`

such implementations are marked as deprecated till we/smb convert this implementation to use AST traversing that is more reliable.
Example of AST based Check: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
attention to `..... extends AbstractJavadocCheck {`

Chance that you/smb will fix this issue in the same design withtout extra regression is not that big.
I could allow this issue, but if on moment of regression we found unwanted/extra violations , I could withdraw approval and reject a fix. So try to fix only you need so much.

All are welcome to rewrite this Check from scratch to use `AbstractJavadocCheck`.

**Comment by cushon** [comment]
I think the regex-based approach isn't the issue here. It's correctly extracting e.g. `Thread.State` from the javadoc comment, but the check compares the references in javadoc to the simple names of imports. Extracting the left-most identifier from qualified names (e.g. `Thread.State` -> `Thread`) fixes this issue and doesn't cause any regressions, see #5046.

**Comment by rnveach** [comment]
Fix was merged
